Sources/PromotedCore/MetricsLogger/MetricsLogger.swift

```diff
@@ -306,16 +306,22 @@ extension MetricsLogger {
   /// Logs a user event.
   ///
   /// Autogenerates the following fields:
+  /// - `userInfo` from `userID` and `logUserID`
   /// - `timing` from `clock.nowMillis`
   ///
   /// - Parameters:
   ///   - properties: Client-specific message
   /// - Returns:
   ///   Logged event message.
   @discardableResult
-  private func logUser(properties: Message? = nil) -> Event_User {
+  private func logUser(properties: Message? = nil) -> Event_User? {
+    if((logUserID ?? "").isEmpty && (userID.stringValue ?? "").isEmpty) {
```

Comment:
Nit: You can also chain optionals with the `guard` statement, which is slightly more idiomatic when looking at edge cases like this:

```
guard userID?.count > 0 || logUserID?.count > 0 else {
  return nil
}
```

Reply:
Also nit: I usually list userID first in the MetricsLogger code